provision/civo.go

```diff
@@ -150,9 +161,33 @@ func (p *CivoProvisioner) lookupID(request HostDeleteRequest) (string, error) {
 	return "", fmt.Errorf("no host with ip: %s", request.IP)
 }
 
+// gets the default network for the selected region.
+func getDefaultNetwork(key, region string) (*Network, error) {
+	apiURL := "https://api.civo.com/v2/networks"
+	values := url.Values{}
+	values.Add("region", region)
+	body, err := apiCall(key, http.MethodGet, apiURL, strings.NewReader(values.Encode()))
+	if err != nil {
+		return nil, err
+	}
+	networks := make([]Network, 0)
+	json.NewDecoder(bytes.NewReader(body)).Decode(&networks)
```

Comment:
Does Decode not return an error?

I would suggest `json.Unmarshal` given that it's used elsewhere and you have a strongly typed struct.

Then also handle the error.

Reply:
okay, np.